src/main/java/com/ltbc/swarmrestapi/service/SwarmRestApiService.java

```diff
@@ -0,0 +1,82 @@
+package com.ltbc.swarmrestapi.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ltbc.swarmrestapi.service.beesinfo.BeesInfoDetails;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+@Service
+public class SwarmRestApiService {
+
+    private static final String OPENAI_URL = "https://api.openai.com/v1/chat/completions";
+
+    @Value("${OPENAI_API_KEY}")
+    private String apiKey;
+    HttpClient client = HttpClient.newHttpClient();
+
+    private static final Logger logger = LoggerFactory.getLogger(SwarmRestApiService.class);
+
+    public BeesInfoDetails sendPromptToOpenAi(String requestPrompt) throws IOException, InterruptedException {
+        if (apiKey == null) {
+            return null; // TODO: fix up these to return an empty BeesInfoDetails obj
+        }
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(OPENAI_URL))
+                .header("Content-Type", "application/json") // Essential for JSON payloads
+                .header("Authorization", "Bearer " + apiKey)
+                .POST(HttpRequest.BodyPublishers.ofString(getJsonReqStr(requestPrompt)))
+                .build();
+
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+        int statusCode = response.statusCode();
+        String responseBody = response.body();
+
+        if (statusCode == 200) {
+            // TODO: maybe move stuff here instead of below to only process when successful
```

Comment:
Yes, let's do that ^

Reply:
This I will handle in another PR / refactor later on